### PR TITLE
chore: Migrate OrganizerDetailPresenter to ViewModel

### LINKED ISSUE #1587: Migrate OrganizerDetailPresenter to ViewModel
**Describe the chore**
Migrate OrganizerDetailPresenter to ViewModel.
Writing unit tests for the ViewModel.
 
**Would you like to work on the issue?**
- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by ShridharGoel**
Fixes #1587 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream development branch.

Changes: 

Migrated OrganizerDetailPresenter to ViewModel.
Added unit tests for the ViewModel.

### CODE REVIEW

File: app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailViewModel.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    private User user;
+
+    private final SingleEventLiveData<Boolean> progress = new SingleEventLiveData<>();
+    private final SingleEventLiveData<String> error = new SingleEventLiveData<>();
+    private final SingleEventLiveData<String> success = new SingleEventLiveData<>();
+    private final SingleEventLiveData<User> userLiveData = new SingleEventLiveData<>();
+
+    @Inject
+    public OrganizerDetailViewModel(UserRepositoryImpl userRepository, AuthService authService) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailViewModel.java]: Use the interface, not Impl

Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailViewModel.java]: Done

File: app/src/test/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailViewModelTest.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+@RunWith(JUnit4.class)
+public class OrganizerDetailViewModelTest {
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+    @Rule
+    public TestRule rule = new InstantTaskExecutorRule();
+
+    @Mock
+    private UserRepositoryImpl userRepository;
```
Comment (iamareebjamal) [inline app/src/test/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailViewModelTest.java]: Same, use interface

Comment (ShridharGoel) [reply app/src/test/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailViewModelTest.java]: Done

**Pull request merged**